apps/avifdec.c

```diff
@@ -159,11 +159,12 @@ int main(int argc, char * argv[])
             ignoreICC = AVIF_TRUE;
         } else if (!strcmp(arg, "--size-limit")) {
             NEXTARG();
-            imageSizeLimit = strtoul(arg, NULL, 10);
-            if ((imageSizeLimit > AVIF_DEFAULT_IMAGE_SIZE_LIMIT) || (imageSizeLimit == 0)) {
+            unsigned long value = strtoul(arg, NULL, 10);
+            if ((value > AVIF_DEFAULT_IMAGE_SIZE_LIMIT) || (value == 0)) {
```

Comment:
If we pass the test here, we know `value <= AVIF_DEFAULT_IMAGE_SIZE_LIMIT <= UINT32_MAX`, so the `uint32_t` cast at line 167 is safe.